magda-web-client/src/UI/Pagination.js

```diff
@@ -44,90 +44,85 @@ class Pagination extends Component {
     }
 
     renderPageList(max, current) {
-        const pages = [...Array(max).keys()].map(x => ++x);
-        const margins = [...Array(3).keys()].map(x => ++x);
-        if (max > 5) {
-            if (current <= 3 || current === max) {
-                return (
-                    <ul className="pagination-list">
-                        {current > 1 && this.renderPrevButton(current)}
-                        {margins.map(i => (
-                            <li key={i}>
-                                <button
-                                    onClick={this.onClick.bind(this, i)}
-                                    className={`${
-                                        i === current
-                                            ? "current"
-                                            : "non-current"
-                                    }`}
-                                >
-                                    {i}
-                                </button>
-                            </li>
-                        ))}
-                        <li>{this.renderDisabledButton()}</li>
-                        <li>
-                            <button
-                                onClick={this.onClick.bind(this, max)}
-                                className={`${
-                                    max === current ? "current" : "non-current"
-                                }`}
-                            >
-                                {max}
-                            </button>
-                        </li>
-                        {current < max && this.renderNextButton(current)}
-                    </ul>
-                );
+        //--- make sure values always stay in range
+        if (!(current >= 1)) current = 1;
+        if (!(max >= current)) {
+            if (max >= 1) current = max;
+            else max = current;
+        }
+
+        //-- Rule 1: detail see my issue comment on github
```

Comment:
If someone comes along in 6 months and tries to modify this it'll be really hard to find the issue on Github - can we just copy the rules into the comments?